libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp

```diff
@@ -0,0 +1,43 @@
+#include "AP_BattMonitor_SMBus.h"
+
+#define AP_BATTMONITOR_SMBUS_PEC_POLYNOME 0x07 // Polynome for CRC generation
+
+/// get_PEC - calculate packet error correction code of buffer
+uint8_t AP_BattMonitor_SMBus::get_PEC(const uint8_t i2c_addr, uint8_t cmd, bool reading, const uint8_t buff[], uint8_t len) const
+{
+    // exit immediately if no data
+    if (len <= 0) {
```

Comment:
GitHub only allows me to put one +1, otherwise I'd put a lot more :slightly_smiling_face: